olive/cli/quantize.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+# ruff: noqa: T201
+# ruff: noqa: RUF012
+
+import tempfile
+from argparse import ArgumentParser
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, Dict
+
+from olive.cli.base import (
+    BaseOliveCLICommand,
+    add_accelerator_options,
+    add_dataset_options,
+    add_input_model_options,
+    add_logging_options,
+    add_remote_options,
+    is_remote_run,
+    update_accelerator_options,
+    update_dataset_options,
+    update_input_model_options,
+)
+from olive.common.utils import hardlink_copy_dir, set_nested_dict_value
+
+
+class QuantizeCommand(BaseOliveCLICommand):
+    _CONFIG_TEMPLATE = {
+        "input_model": {"type": "HfModel", "load_kwargs": {"attn_implementation": "eager"}},
+        "systems": {
+            "local_system": {
+                "type": "LocalSystem",
+                "accelerators": [{"device": "gpu", "execution_providers": ["CUDAExecutionProvider"]}],
+            }
+        },
+        "data_configs": [
+            {
+                "name": "default_data_config",
+                "type": "HuggingfaceContainer",
+                "load_dataset_config": {},
+                "pre_process_data_config": {},
+                "dataloader_config": {},
+                "post_process_data_config": {},
+            }
+        ],
+        "passes": {
+            "awq": {"type": "AutoAWQQuantizer"},
+            "gptq": {
+                # Ref: https://github.com/AutoGPTQ/AutoGPTQ/pull/651/files
+                "type": "GptqQuantizer",
+                "data_config": "default_data_config",
+            },
+            "quarot": {
+                "type": "QuaRot",
+                "w_rtn": True,
+                "rotate": True,
+                "w_bits": 4,
+                "a_bits": 4,
+                "k_bits": 4,
+                "v_bits": 4,
+                "calibration_data_config": None,
+            },
+        },
+        "pass_flows": [],
+        "output_dir": "models",
+        "host": "local_system",
+        "target": "local_system",
+    }
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "quantize",
+            help="Quantize the input model",
+        )
+
+        # model options
+        add_input_model_options(sub_parser, enable_hf=True, enable_pt=True, default_output_path="quantized-model")
+
+        # Logging options
+        add_logging_options(sub_parser)
+
+        sub_parser.add_argument(
+            "--algorithms",
+            type=str,
+            nargs="*",
+            required=True,
+            choices=sorted(QuantizeCommand._CONFIG_TEMPLATE["passes"].keys()),
+            help="List of quantization algorithms to run.",
+        )
+
+        # dataset options
+        add_dataset_options(sub_parser)
```

Comment:
the train/eval names in this doesn't make sense in this scenario. Maybe the option to just have just one `split`,`subset` or one each for train/eval can be a parameter for this function like in `add_input_model_options`